ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -292,24 +305,19 @@ namespace ix
             {
                 ssize_t ret = _socket->recv((char*)&_readbuf[0], _readbuf.size());
 
-                if (ret < 0 && (_socket->getErrno() == EWOULDBLOCK ||
-                                _socket->getErrno() == EAGAIN))
+                if (ret < 0 && _readyState != CLOSING && (_socket->getErrno() == EWOULDBLOCK ||
+                                                          _socket->getErrno() == EAGAIN))
                 {
                     break;
                 }
                 else if (ret <= 0)
                 {
-                    _rxbuf.clear();
```

Comment:
We are not clearing that buffer anymore in any place now, which is a problem. (_rxbuf.clear())

Reply:
Yes